Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -80,12 +80,12 @@ void NetIPTopDevice::DoState(PointerWrap& p)
   Device::DoState(p);
 }
 
-static int inet_pton(const char* src, unsigned char* dst)
+static std::optional<u32> inet_pton(const char* src)
 {
   int saw_digit = 0;
   int octets = 0;
-  unsigned char tmp[4]{};
-  unsigned char* tp = tmp;
+  u32 tmp;
+  unsigned char* tp = reinterpret_cast<unsigned char*>(&tmp);
```

Comment:
I think this would be cleaner as `std::array<unsigned char, 4> tp{};` followed by a `return BitCast<u32>(tp)` at the end of the function?

Reply:
I've turned `tmp` into an `std::array`. But I've kept `tp` as a pointer, since the code increments that pointer to keep track of where it is.